### PR TITLE
Implement Data.Parser.ParserD.takeBetween

### PR DESCRIPTION
**Pull request by specdrake**

### CODE REVIEW

**Comment by harendra-kumar** [comment]
Please use one PR per combinator. Do not add multiple in one. It is easier to review and merge a PR with a single combinator.

**Comment by harendra-kumar** [comment]
Is this ready for review, the status says "Draft"?

**Comment by specdrake** [comment]
> Please use one PR per combinator. Do not add multiple in one. It is easier to review and merge a PR with a single combinator.

Okay

**Comment by specdrake** [comment]
> Is this ready for review, the status says "Draft"?

I was planning to add all the combinators in this PR, I'll separate them into multiple PRs and add its tests and benchmarks in the current PR.

File: src/Streamly/Internal/Data/Parser.hs [hunk 1]
```diff
@@ -337,8 +337,30 @@ either = K.toParserK . D.either
 -- | @takeBetween m n@ takes a minimum of @m@ and a maximum of @n@ input
 -- elements and folds them using the supplied fold.
 --
--- Stops after @n@ elements.
--- Fails if the stream ends before @m@ elements could be taken.
+-- Stops after @m@ elements.
```
[the remark this answers is not in the document]
Comment (specdrake) [reply src/Streamly/Internal/Data/Parser.hs:340]: Yes, it should be `n`.

File: src/Streamly/Internal/Data/Parser.hs [hunk 2]
```diff
@@ -337,8 +337,30 @@ either = K.toParserK . D.either
...
+-- Fails if the stream ends before @n@ elements could be taken.
```
[the remark this answers is not in the document]
Comment (specdrake) [reply src/Streamly/Internal/Data/Parser.hs:341]: Yes, I switched the too.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Mostly good. Rebase it and take care of some nitpicks.

File: test/Streamly/Test/Data/Parser/ParserD.hs [hunk 3]
```diff
@@ -146,6 +146,36 @@ satisfy =
             predicate = (>= mid_value)
 
 -- Sequence Parsers Tests
+takeBetweenPass :: Property
+takeBetweenPass =
```
[the remark this answers is not in the document]
Comment (specdrake) [reply test/Streamly/Test/Data/Parser/ParserD.hs:150]: Yes, I was thinking there are many duplicate tests.

**Comment by harendra-kumar** [comment]
Merged offline.

**Pull request closed without merging**